.github/workflows/lint.yaml

```diff
@@ -0,0 +1,25 @@
+name: Lint
+
+on:
+  pull_request:
+  push:
+    branches:
+      - main
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  lint:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+      - uses: actions/setup-python@v5
+        with:
+          python-version: '3.10'
+      - name: Lint
```

Comment:
Could we use https://github.com/pre-commit/action instead? It's already added to the allow-list for 3rd party actions.

Reply:
From https://github.com/pre-commit/action:

> this action is in maintenance-only mode and will not be accepting new features.
>
> generally you want to use [pre-commit.ci](https://pre-commit.ci/) which is faster and has more features.

Could we use pre-commit.ci instead?